src/OrchardCore/OrchardCore.DisplayManagement/Razor/RazorPage.cs

```diff
@@ -80,6 +80,35 @@ public IShapeFactory Factory
         /// </summary>
         /// <param name="shape">The shape.</param>
         public Task<IHtmlContent> DisplayAsync(dynamic shape)
+        {
+            EnsureDisplayHelper();
+
+            if (shape is IShape s)
+            {
+                return _displayHelper.ShapeExecuteAsync(s);
+            }
+            else
+            {
+                if (shape is IHtmlContent hc)
+                {
+                    return Task.FromResult(hc);
+                }
+                else if (shape is string str)
+                {
+                    return Task.FromResult<IHtmlContent>(new HtmlString(HtmlEncoder.Encode(str)));
```

Comment:
See my suggestion in the other comment or at least here use new `StringHtmlContent(str)` in place of encoding here

And maybe replace the `else` and `else if` blocks by only `if` blocks as they always returns something ;)

Also not sure about throwing below, regarding what we do elsewhere, returning an empty html string

Reply:
Ok for the suggestions, but I want to throw here. I/we need to add a comment to mention this should go away eventually, and that's why we throw. We can make it obsolete once we know we have removed the other usages of this method.

Reply:
Okay cool then